app/routes/tasks.py

```diff
@@ -0,0 +1,128 @@
+import requests
+import os
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response(
+            {"message": f"{cls.title} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response(
+            {"message": f"{cls.title} {model_id} not found"}, 404))
+    
+    return model
+
+def check_for_missing_data(request_body):
```

Comment:
You're on a the right track here! Notice you have a similar function in the goal routes. You could make this more generic and move it to a helpers file. If you change the argument signature to accept a list of params you want to check for, you could use that instead of hard-coding the values like on line 25.